app/src/main/java/com/example/cristiana/weatherapp/OneDayActivity.java

```diff
@@ -0,0 +1,160 @@
+    Button mDaysButton;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_oneday);
+
+        /* get IDs of the elements from layout */
+        getParameters();
+        /* get the data from web */
+        updateInfo();
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v.getId() == R.id.daysButton) {
+            Intent intent = new Intent(this, TenDaysActivity.class);
+            intent.putExtra(LOC_SAVED, mLocationName);
+            startActivity(intent);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater menuInflater = getMenuInflater();
+        menuInflater.inflate(R.menu.oneday_menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.Return) {
+            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+            preferences.edit().remove(LOC_SAVED);
+
+            Intent intent = new Intent(this, MainActivity.class);
+            startActivity(intent);
+            finish();
+
+            return true;
+        }
+
+        return false;
+    }
+
+    private void getParameters() {
+        mLocationName = getIntent().getStringExtra(LOC_SAVED);
+        mLocationValue = (TextView) findViewById(R.id.locationValue);
+        mTemperatureValue = (TextView) findViewById(R.id.temperatureValue);
+        mDescriptionValue = (TextView) findViewById(R.id.descriptionValue);
+        mHumidityValue = (TextView) findViewById(R.id.humidityValue);
+        mPressureValue = (TextView) findViewById(R.id.pressureValue);
+        mDaysButton = (Button) findViewById(R.id.daysButton);
+
+        String locName = mLocationName.substring(0, 1).toUpperCase() + mLocationName.substring(1);
+        mLocationName = locName;
+
+        mDaysButton.setOnClickListener(this);
+    }
+
+    private String errorMessage(int code) {
+        String message = "An error occured";
+        switch (code) {
+            case 401: message = "Invalid location"; break;
+            case 404: message = "Page not found"; break;
+            case 500: message = "Internal server error"; break;
+            case 503: message = "Serice unavailable"; break;
+            case 550: message = "Permission denied"; break;
+        }
+
+        return message;
+    }
+
+    private String failureMessage(Throwable t) {
+        String message = "Connection error";
+
+        if (t instanceof UnknownHostException)
+            message = "No Internet connecion found";
+
+        if (t instanceof TimeoutException)
+            message = "Connection time expired";
+
+        return message;
+    }
+
+    private void updateInfo() {
+        /* show the searched location */
+        String key = "61ae4591669ba9f38b46d26e6ee808d5";
```

Comment:
Don't keep your appid spread out all over the app. Do the following:

1. Inside of your retrofit interface define a string constant that holds this appid
2. Instead of passing the appid as a key parameter to each call, in your singleton constructor for the retrofit interface add an application-level interceptor that does that for you